nominal/core/connection.py

```diff
@@ -0,0 +1,113 @@
+        )
+
+    def _get_series_achetypes_paginated(self) -> Iterable[datasource_api.ChannelMetadata]:
+        request = datasource_api.SearchChannelsRequest(
+            data_sources=[self.rid], exact_match=[], fuzzy_search_text="", previously_selected_channels={}
+        )
+        while True:
+            response = self._clients.datasource.search_channels(self._clients.auth_header, request)
+            for channel_metadata in response.results:
+                if channel_metadata.series_rid.type == "seriesArchetype":
+                    yield channel_metadata
+                else:
+                    logging.debug("ignoring channel with series_rid type %s", channel_metadata.series_rid.type)
+            if response.next_page_token is None:
+                break
+            request = datasource_api.SearchChannelsRequest(
+                data_sources=[self.rid],
+                exact_match=[],
+                fuzzy_search_text="",
+                previously_selected_channels={},
+                page_token=response.next_page_token,
+            )
+
+    def _resolve_archetypes_to_channels(self, channel_names: Sequence[str]) -> Iterable[Channel]:
+        """Given archetype names ("shirts", "sweaters") and existing tag values ({color: [red, green], size: [S, M, L]}),
+        resolve all possible combinations of archetype names and tag values to logical series. We will try to resolve
+        the following logical series:
+            * shirts, {color: red, size: S}
+            * shirts, {color: red, size: M}
+            * shirts, {color: red, size: L}
+            * shirts, {color: green, size: S}
+            * shirts, {color: green, size: M}
+            * shirts, {color: green, size: L}
+            * sweaters, {color: red, size: S}
+            * sweaters, {color: red, size: M}
+            * sweaters, {color: red, size: L}
+            * sweaters, {color: green, size: S}
+            * sweaters, {color: green, size: M}
+            * sweaters, {color: green, size: L}
+        """
+        req = timeseries_logicalseries_api.BatchResolveSeriesRequest(
+            requests=[
+                timeseries_logicalseries_api.ResolveSeriesRequest(datasource=self.rid, name=name, tags=tags)
+                for name in channel_names
+                for tags in _tag_product(self._tags)
+            ]
+        )
+        resp = self._clients.logical_series.resolve_batch(self._clients.auth_header, req)
+        for resolved_series in resp.series:
+            if resolved_series.type == "error" and resolved_series.error is not None:
+                raise RuntimeError(f"error resolving series: {resolved_series.error}")
+            # TODO(alkasm): is there a batch get_logical_series ?
+            series = self._clients.logical_series.get_logical_series(self._clients.auth_header, resolved_series.rid)
+            yield Channel._from_conjure_logicalseries_api(self._clients, series)
+
+    def _get_channels(self) -> Iterable[Channel]:
```

Comment:
this is currently "private" and unused; I'm not 100% sure if this implementation is correct, and making a distinct `get_logical_series` call per channel is not what we want to do. Still I wanted this code to stick around for now so we know how to model it in the future, the tag resolution was particularly confusing so don't want to erase the context.